app/controllers/locations_controller.rb

```diff
@@ -17,7 +17,13 @@ def create
   end
 
   def edit
-    @location = Location.find(params[:id])
+    location = Location.find(params[:id])
```

Comment:
Whenever you're running the same code in both the if and the else, it means that you could run it outside of the conditional instead. I'd suggest
```
def edit
    @location = Location.find(params[:id])
    if !current_user.id == location.user_id
      redirect_to @location, notice: 'You are not authorized to edit this DateSpot!'
    end
end
```